names/module.py

```diff
@@ -0,0 +1,250 @@
+from typing import Optional
+
+import discord
+from discord.ext import commands
+
+from pie import check, utils, i18n, logger
+from modules.base.admin.database import BaseAdminModule
+from modules.boards.karma.database import KarmaMember
+
+from .database import Price
+
+_ = i18n.Translator("modules/fun").translate
+bot_log = logger.Bot.logger()
+
+
+class Names(commands.Cog):
+    """Pay by karma and I change your name."""
+
+    def __init__(self, bot: commands.Bot):
+        # Check if dependency module is loaded.
+        if not BaseAdminModule.get("boards.karma").enabled:
+            raise Exception(
+                "Unable to load names module due to missing module `boards.karma`."
+            )
+
+        Price.add_commnand("set", 500)
+        Price.add_commnand("unset", 200)
+        self.bot = bot
+
+    @commands.guild_only()
+    @check.acl2(check.ACLevel.MEMBER)
+    @commands.cooldown(rate=5, per=20.0, type=commands.BucketType.user)
+    @commands.command()
+    async def prices(self, ctx):
+        """Display prices for various services"""
+
+        embed = utils.discord.create_embed(
+            author=ctx.author,
+            title=_(ctx, "Change of nickname"),
+            description=_(
+                ctx,
+                "Values are in karma points. Use commnand **nickname/name** for more info.",
+            ),
+        )
+
+        embed.add_field(
+            name=_(ctx, "Set"),
+            value=_(ctx, "Price {price}").format(price=Price.get("set").price),
+        )
+        embed.add_field(
+            name=_(ctx, "Reset"),
+            value=_(ctx, "Price {price}").format(price=Price.get("unset").price),
+        )
+
+        await ctx.reply(embed=embed)
+
+    @commands.bot_has_permissions(manage_nicknames=True)
+    @commands.guild_only()
+    @check.acl2(check.ACLevel.MEMBER)
+    @commands.cooldown(rate=5, per=20.0, type=commands.BucketType.user)
+    @commands.group(name="nickname", aliases=["name"])
+    async def nickname_(self, ctx):
+        """Change your nickname"""
+        await utils.discord.send_help(ctx)
+
+    @commands.bot_has_permissions(manage_nicknames=True)
+    @commands.guild_only()
+    @check.acl2(check.ACLevel.MEMBER)
+    @commands.cooldown(rate=5, per=20.0, type=commands.BucketType.user)
+    @nickname_.command(name="set")
+    async def set_nickname(self, ctx, nickname: Optional[str] = None):
+        """Change server nickname"""
+        if not BaseAdminModule.get("boards.karma").enabled:
+            await bot_log.error(
+                ctx.author,
+                ctx.channel,
+                "Repository boards.karma is not loaded. Cannot set nickname.",
+            )
+            await ctx.reply(
+                _(ctx, "Cannot change nickname because mudule karma is not loaded.")
+            )
+            return
+        if nickname is None:
+            await ctx.reply(_(ctx, "Unless you give me nickname I cant't do nothing."))
+            return
```

Comment:
If you use `nickname: str` in the function argument (without `= None`) you don't have to handle this.